Makefile

```diff
@@ -219,6 +231,13 @@ node-agent: ${ISTIO_BIN}/node_agent
 .PHONY: pilot
 pilot: ${ISTIO_BIN}/pilot-discovery
 
+# istioctl-all makes all of the non-static istioctl executables for each supported OS
+.PHONY: istioctl-all
+istioctl-all: $(ISTIOCTL_LIST)
+
+# provide an alias for each istioctl-OS that's just the filename
+$(foreach ITEM,$(ISTIOCTL_LIST),$(eval .PHONY: $(ITEM)) $(eval $(ITEM): ${OUT}/$(ITEM)))
```

Comment:
Yeah, the more I look at this, the more uncomfortable I feel about these kinds of things in the makefiles. Makefiles can get worse if we throw in all its magic.. 
Something simple like
`istioctl: istioctl-osx  istioctl-win istioctl-linux` is pretty simple to read and understand. It is fine if in that process we miss one or two OSes.. People will come and add another entry to this list.